WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java

```diff
@@ -359,33 +359,32 @@ public static boolean isFollowing(
         }
     }
 
-    public static ReaderTag getValidTagForSharedPrefs(
-            ReaderTag tag,
+    public static @NonNull ReaderTag getValidTagForSharedPrefs(
+            @NonNull ReaderTag tag,
             boolean isTopLevelReader,
             FilteredRecyclerView recyclerView
     ) {
-        ReaderTag validTag = tag;
+        if (!isTopLevelReader) {
+            return tag;
+        }
 
-        if (isTopLevelReader) {
-            if (tag.isDiscover() || tag.isPostsILike() || tag.isBookmarked()
-                    ||
-                (recyclerView != null && recyclerView.isValidFilter(tag))
-            ) {
-                validTag = tag;
+        boolean isValidFilter = (recyclerView != null && recyclerView.isValidFilter(tag));
+        boolean isSpecialTag = tag.isDiscover() || tag.isPostsILike() || tag.isBookmarked();
```

Comment:
I think combining these into a single `boolean` is very helpful here, but I wasn't sure what a good name for it would be. These seemed like special tags, so I called it that. If there is a better name for this collection, we should use that.

Reply:
Correct! Just to add a bit of context, those tags are a `special` kind of valid filters (that are all the filters hosted in the main tabs filtering). So `isSpecialTag` seems a good fit.